zephyr/esp32/src/wifi/esp_wifi_adapter.c

```diff
@@ -966,7 +1016,6 @@ esp_err_t esp_wifi_init(const wifi_init_config_t *config)
 	return result;
 }
 
-static void esp_wifi_free(void *mem)
-{
-	k_free(mem);
-}
+#ifdef CONFIG_ESP32_WIFI_HEAP_MEM_POOL
+SYS_INIT(esp_wifi_heap_init, PRE_KERNEL_2, 50);
```

Comment:
I suggest replacing the hard-coded `50` level to `CONFIG_KERNEL_INIT_PRIORITY_DEVICE` instead.

Reply:
fixed